File: AbstractFactory/Java/Product Factory/Creator2.java
public class Creator2 extends AbstractCreator {
    //只生產產品等級為2的A產品
    public AbstractProductA createProductA() {
        return new ProductA2();
    }
    //只生產產品等級為2的B產品
    public AbstractProductB createProductB() {
        return new ProductB2();
    }
}

File: AbstractFactory/PHP/Creator2.php
<?php

include_once ('AbstractCreator.php');
include_once ('ProductA2.php');
include_once ('ProductB2.php');
class Creator2 extends AbstractCreator
{

    public function createProductA(): AbstractProductA
    {
        // TODO: Implement createProductA() method.
        return new ProductA2();
    }

    public function createProductB(): AbstractProductB
    {
        // TODO: Implement createProductB() method.
        return new ProductB2();
    }
}